org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -110,6 +112,13 @@ import org.eclipse.xtext.validation.Issue
 		if (params.rootPath === null) {
 			throw new IllegalArgumentException("Bad initialization request. rootPath must not be null.")
 		}
+		val rootPath = Paths.get(params.rootPath)
+		if (!Files.exists(rootPath) || !Files.isDirectory(rootPath)) {
+			throw new IllegalArgumentException("Bad initialization request. rootPath does not exist or is not a directory.")
+		}
+		if (!Files.isWritable(rootPath)) {
```

Comment:
You're right - good point.
Maybe I had a slightly wrong idea of the language server.
Is the LS allowed or in charge of creating files? Or is that dann by another component?